docs/guides/using_TransformAnimationComponent.rst

```diff
@@ -0,0 +1,160 @@
+===================================================
+Animating Objects using TransformAnimationComponent
+===================================================
+
+TransformAnimationComponent can be used to easily animate scenes which have to be animated endlessly without relying on scripting for the same.
+
+It is quite easy to as it uses a keyframe based interface to achieve this.
+
+Setting up
+----------
+
+Create a new scene and assign the TransformComponent, ModelComponent and TransformAnimationComponent to it.
+
+Creating a new scene:
+
+.. figure:: images/AnimTestCreate.png
+
+Adding the Components by right clicking on the scene in inspector:
+
+.. figure:: images/AnimTestComp.png
+
+Select the EditorCamera:
+
+.. figure:: images/AnimTestCam.png
+
+Set the camera to EditorCamera and position it such that it has a good view of the scene.
+
+.. figure:: images/AnimTestView.png
+
+Select the EditorCamera as default camera for the scene.
+
+.. figure:: images/AnimTestDefCam.png
+
+Interface
+---------
+
+* Transition Type : How the transition must be between keyframes. Explaining the options:
+
+   * SmashSmash : Abrupt start and end transition
+   * EaseEase : Smooth start and end transition
+   * SmashEase : Abrubt start and smooth end transition
+   * EaseSmash : Smooth start and abrupt end transition
+
+* Animation Mode : How the animation should play. Explaining the options:
+
+   * Looping : Animation plays from Start time to End time unidirectionally.
+   * Alternating : Animation plays from Start time to EndTime and then reverses from End time to Start time.
+
+* Time : Shows the time while animation plays
+
+* Reset : resets the animation
+
+* Play in Editor : Play the animation in editor
+
+* Play on start : Play the animation when game starts
+
+* Keyframes : To set the animation keyframes
+
+  - Set Keyframe : To add a new keyframe at the end
+  - Pop Keyframe : To remove the last keyframe
```

Comment:
Bold the different labels you are explaining. ie - (Time, Reset, Play in Editor, Animation Mode, Looping, etc). Without it, it's looking a little unstructured. 
![image](https://user-images.githubusercontent.com/54415525/156161502-be14dee7-6457-4a14-af22-e55c3fcd8eb0.png)